src/FreeDSx/Snmp/Message/Security/SecurityParametersInterface.php

```diff
@@ -25,4 +25,6 @@ interface SecurityParametersInterface extends ProtocolElementInterface
      * @return int
      */
     public function getSecurityModel() : int;
+
+    public function getUsername() : string;
```

Comment:
It's not a given that all SNMP security models will have a username field. The UserSecurityModel does, but other ones do not. So I would not like to enforce it on the interface. Is it possible to do `instanceof` checks for the UserSecurityParameters class instead where applicable?

Reply:
What about now?